src/glob.zig

```diff
@@ -1636,7 +1649,7 @@ pub fn GlobWalker_(
                                 end_byte,
                                 has_relative_patterns,
                             )) |component| {
-                                saw_special = saw_special or component.syntax_hint.isSpecialSyntax();
+                                saw_special = if (!saw_special) component.syntax_hint.isSpecialSyntax() else saw_special;
```

Comment:
I think this change is confusing - the `or` was more readable and did the same thing